Update README quickstart (#36)
ngrok/ngrok-python: README.md

If you're looking for the previous agent downloader project, it's over [here](https://github.com/OpenIoTHub/ngrok).

# Documentation

A quickstart guide and a full API reference are included in the [ngrok-python API documentation](https://ngrok.github.io/ngrok-python/).

# Quickstart

1. Install the `ngrok-python` package from [PyPI](https://pypi.org/project/ngrok) using `pip`:

```shell
python -m pip install ngrok
```

> 2. After you've installed the package, you'll need an Auth Token. Retrieve one on the

Reviewer: Let's change the verbiage for "Auth Token" to "authtoken" to match how we describe it in the product and documentation. Let's make sure this is reflected throughout the entire readme.

Author: Have taken care of this in latest commit.

[Auth Token page of your ngrok dashboard](https://dashboard.ngrok.com/get-started/your-authtoken).

3. Add the following code block using the [connect method](https://github.com/ngrok/ngrok-python/blob/main/examples/ngrok-connect-minimal.py) to expose your python application at port `9000` on `localhost`:

```python
import ngrok
tunnel = ngrok.connect(9000, authtoken_from_env=True)
print (f"Ingress established at {tunnel.url()}")
```

> There are many more examples in [the /examples directory](https://github.com/ngrok/ngrok-python/tree/main/examples).

Reviewer: ```suggestion
You can find more examples in [the /examples directory](https://github.com/ngrok/ngrok-python/tree/main/examples).
```

## Authorization

To use most of ngrok's features, you'll need an authtoken. To obtain one, sign up for free at [ngrok.com](https://dashboard.ngrok.com/signup) and retrieve it from the [authtoken page in your ngrok dashboard](https://dashboard.ngrok.com/get-started/your-authtoken). Once you have copied your authtoken, you can reference it in several ways.

You can set it in the `NGROK_AUTHTOKEN` environment variable and pass `authtoken_from_env=True` to the [connect](https://ngrok.github.io/ngrok-python/module.html) method:

```python
ngrok.connect(authtoken_from_env=True, ...)
```

Or pass the authtoken directly to the [connect](https://ngrok.github.io/ngrok-python/module.html) method:

```python
ngrok.connect(authtoken=token, ...)
```

> Or set it for all connections with the [set_auth_token](https://ngrok.github.io/ngrok-python/module.html) method:

Author: This mirrors the community library's name, since Russ would like to make migration easier.

> Or set it for all connections with the [set_auth_token](https://ngrok.github.io/ngrok-python/module.html) method:

Author: This will be dealt with in a future sdk api revamp.

python -m pip install ngrok
```

To verify that the library is correctly installed use the following code, which forwards to `localhost` port `80`:
```python
import ngrok
tunnel = ngrok.connect()
```

`ngrok-python` officially supports Python 3.7+.

# Documentation

A quickstart guide and a full API reference are included in the [ngrok-python API documentation](https://ngrok.github.io/ngrok-python/).

# Quickstart

After you've installed the package, you'll need an Auth Token. Retrieve one on the
[Auth Token page of your ngrok dashboard](https://dashboard.ngrok.com/get-started/your-authtoken)

> Here is a minimal code block [using the 'connect' convenience function](https://github.com/ngrok/ngrok-python/blob/main/examples/ngrok-connect-minimal.py), with `authtoken_from_env=True` to use an Auth Token from the `NGROK_AUTHTOKEN` environment variable, and making a connection to `localhost` port `9000`:

Author: This should be solved with latest commit.

```python
tunnel = ngrok.connect(9000, authtoken_from_env=True)
print (f"Ingress established at {tunnel.url()}")
```

There are multiple examples in [the /examples directory](https://github.com/ngrok/ngrok-python/tree/main/examples).

## Authorization

To use most features of ngrok, you need to obtain an Auth Token. You can get an Auth Token by signing up for free at [ngrok.com](https://dashboard.ngrok.com/signup) and then retrieving it from the [Auth Token page of your ngrok dashboard](https://dashboard.ngrok.com/get-started/your-authtoken). Once you have an Auth Token, you can reference it in several ways.

You can set the authtoken in the environment variable `NGROK_AUTHTOKEN` and then pass `authtoken_from_env=True` to the [connect](https://ngrok.github.io/ngrok-python/module.html) function:

```python
ngrok.connect(authtoken_from_env=True, ...)
```

> You can also set the default Auth Token to use for all connections by calling the [set_auth_token](https://ngrok.github.io/ngrok-python/module.html) function:

Author: This should be solved with latest commit.

```python
ngrok.set_auth_token(token)
```

Or the Auth Token can be passed directly to the [connect](https://ngrok.github.io/ngrok-python/module.html) function:

```python
ngrok.connect(authtoken=token, ...)
```

## Connection

The [connect](https://ngrok.github.io/ngrok-python/module.html) function is the most common way to use this library. It will start an ngrok session if it is not already running, and then establish a tunnel to the specified address. If there is an asynchronous runtime running the [connect](https://ngrok.github.io/ngrok-python/module.html) function returns a promise that resolves to the public tunnel object.

> With just an integer, the `connect` function will forward to `localhost` on the specified port, or the host can be specified as a string:

Author: With no arguments it defaults to `localhost` port `80`, mimicking the community library behavior. That is mentioned in the old `Installation` section, but since that is being removed I'll add an example line here.

```

The second (optional) argument is the tunnel type, with defaults to `http`. To create a TCP tunnel:

```python
tunnel = ngrok.connect(25565, "tcp")
```

Since the options are kwargs, you can also use the `**` operator to pass a dictionary of configuration:

```python
options = {"authtoken_from_env":True, "response_header_add":"X-Awesome:yes"}
tunnel = ngrok.connect(8080, **options)
```

## Disconnection

To close a tunnel use the [disconnect](https://ngrok.github.io/ngrok-python/module.html) function with the `url` of the tunnel to close. If there is an asynchronous runtime running the [disconnect](https://ngrok.github.io/ngrok-python/module.html) function returns a promise that resolves when the call is complete.

```python
ngrok.disconnect(url)
```

Or omit the URL to close all tunnels:

```python
ngrok.disconnect()
```

The [close](https://ngrok.github.io/ngrok-python/ngrok_tunnel.html) method on a tunnel will shut it down, and also stop the ngrok session if it is no longer needed. This method returns a promise that resolves when the tunnel is closed.

```python
await tunnel.close()
```

## Listing Tunnels

> To list all current non-closed tunnels use the [get_tunnels](https://ngrok.github.io/ngrok-python/module.html) function. If there is an asynchronous runtime running the [get_tunnels](https://ngrok.github.io/ngrok-python/module.html) function returns a promise that resolves to the list of tunnel objects.

Author: It's funny, I started with `method` then switched the global static ones to `function` to try and be more precise. But it does feel more natural this way.
